src/features/assistant/MozAssistantChat.ts

```diff
@@ -0,0 +1,253 @@
+        if (!isSearchButton) {
+          const next = await getQuickPromptsInConversation(this.messages, 2)
+          this.quickPrompts = next
+        }
+      } catch (e) {
+        console.warn('[assistant][quick-prompts] in-convo generation failed:', e)
+      }
+      this.updateComplete.then(() => this.scrollToBottom())
+    }
+  }
+
+  scrollToBottom() {
+    const el = this.renderRoot?.querySelector('.chat-window') as HTMLElement
+    if (el) el.scrollTo({ top: el.scrollHeight, behavior: 'smooth' })
+  }
+
+  onKeydown(e: KeyboardEvent) {
+    if (e.key === 'Enter' && !e.shiftKey) {
+      e.preventDefault()
+      this.handleSend()
+    }
+  }
+
+  handleClear() {
+    this.messages = []
+    this.inputValue = ''
+    assistantStore.clear()
+    this.refreshQuickPrompts()
+  }
+
+  render() {
+    const renderAssistantContent = (text: string) => {
+      if (text.startsWith('SEARCH_BUTTON:')) {
+        const payload = text.substring('SEARCH_BUTTON:'.length)
+        const [href, label] = payload.split('|')
+        return html`<button class="link-button" @click=${() => browser.tabs.create({ url: href })}>${label || 'Open search'}</button>`
+      }
+      return html`<div class="text" .innerHTML=${this.tinyMark.parse(text)}></div>`
+    }
+    return html`
+      <div class="wrapper">
+        <div class="container">
+          <h3 class="title">Assistant</h3>
+          <div class="chat-window">
+            ${this.messages.map(
+              (m) => html`
+                <div class="bubble-wrapper ${m.role}">
+                  <div class="bubble ${m.role}">
+                    ${renderAssistantContent(m.content)}
+                  </div>
+                </div>`,
+            )}
+            ${this.loading
+              ? html`<div class="loading">Thinking…</div>`
+              : ''}
+          </div>
+
+          ${this.quickPrompts.length
+            ? html`<div class="suggestions">
+                ${this.quickPrompts.map(
+                  (s) =>
+                    html`<button
+                      class="suggestion"
+                      title=${s}
+                      @click=${() => this.handleSend(s)}
+                    >
+                      ${s}
+                    </button>`,
+                )}
+              </div>`
+            : ''}
+
+          <textarea
+            .value=${this.inputValue}
+            @input=${(e: Event) =>
+              (this.inputValue = (e.target as HTMLTextAreaElement).value)}
+            @keydown=${this.onKeydown}
+            placeholder="Type your message…"
+            ?disabled=${this.loading}
+          ></textarea>
+
+          <div class="footer">
+            <button class="secondary-button" @click=${this.handleClear}>
+              Clear
+            </button>
+            <button
+              class="primary-button"
+              @click=${this.handleSend}
```

Comment:
this will pass the click event to handleSend. you probably want `=${() => this.handleSend()}`